unity-renderer/Assets/Scripts/CodeStyleExample.cs

```diff
@@ -0,0 +1,130 @@
+    public enum Side { Left, Right }
+
+    public delegate void Destroction<T> (T current);
+
+    public interface IInitializable
+    {
+        bool IsInitialized { get; }
+
+        void Initialize();
+    }
+
+    [RequireComponent(typeof(Animation))]
+    public class CodeStyleExample : MonoBehaviour, IInitializable
+    {
+        //----------------------------------------------------------------------------------------  Enum, delegates declaration group (protected/private) 
+        private enum Direction { North, South, West, East }
+        private delegate void Interaction<T> (T current);
+        
+        //----------------------------------------------------------------------------------------  Const and static readonly Fields group
+        public const string ASSET_PATH = "AvatarPrefab";
+        private const float MAX_TIMER = 10f;
+
+        public static readonly int WAVE_ANIM_HASH = Animator.StringToHash("Wave");
+        private static readonly int DANCE_ANIM_HASH = Animator.StringToHash("Dance");
+
+        //----------------------------------------------------------------------------------------  Events and UnityEvents group
+        public event Destroction<CodeStyleExample> Destroyed;
+        [NonSerialized] public UnityEvent<bool> Closed; // use [NonSerialized] or [HideInInspector] for UnityEvents
+
+        //----------------------------------------------------------------------------------------  Fields group (other than Const and Static readonly)  
+        public readonly List<int> CachedNumbers = new List<int>();
+        protected readonly List<Vector2> poistions = new List<Vector2>();
+        
+        public int Timer;
+
+        [SerializeField] private AnimationClip[] clips;
+        [SerializeField] private float armor;
+        [SerializeField] private float health;
+        [SerializeField] private bool isVisitble;
+
+        protected float cooldown;
+        
+        private Animation animation;
+        private Interaction<int> interactionsBuffer;
+
+        //----------------------------------------------------------------------------------------  Properties group
+        public bool IsInitialized { get; private set; }
+
+        [PublicAPI]
+        public bool IsVisitble => isVisitble; // [PublicAPI] Public - used only from outside of Unity solution (by other solution)
+
+        //----------------------------------------------------------------------------------------  Public methods group
+        public void Initialize() =>
+            IsInitialized = true;
+
+        public void ApplyDamage(float damage) // Public - called from outside of the class (by other class)
+        {
+            float resultDamage = ComputeDamage(damage);
+            TakeHit(resultDamage);
+        }
+
+        private float ComputeDamage(float damage) => // Called by ApplyDamage
+            damage - armor;
```

Comment:
inline methods, on cases that the implementation is very short, we could do it in the same line:
```
private float ComputeDamage(float damage) => damage -= armor;
```

Reply:
Yeah, it was just more as a scratch. We will discuss it again in Slack more in detail and vote. But I personally like to have experession-body for properties to be on one line, but for methods always put in the new line. In such way it is easier for me to visually separate from properties and methods